kamu-adapter-graphql/src/queries/datasets.rs

```diff
@@ -103,6 +105,206 @@ impl Datasets {
         let account = Account::mock();
         self.by_account_impl(ctx, account, page, per_page).await
     }
+
+    // TODO: Multitenancy
+    /// Creates a new empty dataset
+    #[allow(unused_variables)]
+    async fn create_empty(
+        &self,
+        ctx: &Context<'_>,
+        account_id: AccountID,
+        dataset_kind: DatasetKind,
+        dataset_name: DatasetName,
+    ) -> Result<CreateDatasetResult> {
+        let local_repo = from_catalog::<dyn domain::LocalDatasetRepository>(ctx).unwrap();
+        let dataset_builder = local_repo.create_dataset(&dataset_name).await?;
+
+        // We are generating a key pair and deriving a dataset ID from it.
+        // The key pair is discarded for now, but in future can be used for
+        // proof of control over dataset and metadata signing.
+        let (_keypair, dataset_id) = odf::DatasetID::from_new_keypair_ed25519();
+
+        dataset_builder
+            .as_dataset()
+            .as_metadata_chain()
+            .append(
+                odf::MetadataBlock {
+                    system_time: Utc::now(),
+                    prev_block_hash: None,
+                    sequence_number: 0,
+                    event: odf::MetadataEvent::Seed(odf::Seed {
+                        dataset_id,
+                        dataset_kind: dataset_kind.into(),
+                    }),
+                },
```

Comment:
I prefer if this low-level code would be encapsulated behind a core service. API plugin should work at higher abstraction level.

Reply:
Yeah, was bothering me too - I found a simple solution.